python-client/wmill/wmill/client.py

```diff
@@ -1,527 +1,630 @@
-from typing import Any, Union, Dict
-from typing import Generic, TypeVar, Optional
+from __future__ import annotations
 
-import os
-import json
-from datetime import timedelta
-import logging
 import atexit
+import datetime as dt
+import functools
+import logging
+import os
+import random
 import time
+import warnings
+from json import JSONDecodeError
+from typing import Dict, Any, Union, Literal
+
+import httpx
+
+_client: "Windmill | None" = None
+
+
+logger = logging.getLogger("windmill_client")
+
+JobStatus = Literal["RUNNING", "WAITING", "COMPLETED"]
 
-from time import sleep
-from windmill_api.models.whoami_response_200 import WhoamiResponse200
 
-from windmill_api.client import AuthenticatedClient
+class Windmill:
+    def __init__(self, base_url=None, token=None):
+        base = base_url or os.environ.get("BASE_INTERNAL_URL")
 
-from enum import Enum
+        self.base_url = f"{base}/api"
+        self.token = token or os.environ.get("WM_TOKEN")
+        self.headers = {
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {self.token}",
+        }
+        self.client = self.get_client()
+        self.workspace = os.environ.get("WM_WORKSPACE")
+        self.path = os.environ.get("WM_JOB_PATH")
 
-from windmill_api.types import Unset
+    def get_client(self) -> httpx.Client:
+        return httpx.Client(
+            base_url=self.base_url,
+            headers=self.headers,
+        )
+
+    def get(self, endpoint, raise_for_status=True, **kwargs) -> httpx.Response:
+        endpoint = endpoint.lstrip("/")
+        resp = self.client.get(f"/{endpoint}", **kwargs)
+        if raise_for_status:
+            resp.raise_for_status()
+        return resp
+
+    def post(self, endpoint, raise_for_status=True, **kwargs) -> httpx.Response:
+        endpoint = endpoint.lstrip("/")
+        resp = self.client.post(f"/{endpoint}", **kwargs)
+        if raise_for_status:
+            resp.raise_for_status()
+        return resp
+
+    def create_token(self, duration=dt.timedelta(days=1)) -> str:
+        endpoint = "/users/tokens/create"
+        payload = {
+            "label": f"refresh {time.time()}",
+            "expiration": (dt.datetime.now() + duration).strftime("%Y-%m-%dT%H:%M:%SZ"),
+        }
+        return self.post(endpoint, json=payload).text
+
+    def create_job(
```

Comment:
I was trying to create a distinction since async is often used to denote asynchronous functions in python i.e. those that run in an `asyncio` event loop. Maybe `start_execution`?

Reply:
It's just that we would need to mention "script" or "flow" in such function because the api should be consistent, 

create_script_job or create_job_script would also work